core/src/com/unciv/models/ruleset/unique/UniqueTriggerActivation.kt

```diff
@@ -560,16 +559,17 @@ object UniqueTriggerActivation {
                 if (unitsToPromote.isEmpty()) return null
 
                 return {
-                    val promotedUnitLocations: MutableList<Vector2> = mutableListOf()
+                    val promotedUnits: MutableList<MapUnit> = mutableListOf()
                     for (civUnit in unitsToPromote) {
+                        if (promotionName in civUnit.promotions.promotions) continue
                         civUnit.promotions.addPromotion(promotionName, isFree = true)
```

Comment:
Maybe addPromotion should return a boolean?
this is fine, just thinking aloud 🤔